### PR TITLE
Add prettier formatting for error rep's stacks

### PR DESCRIPTION
**Pull request by gregtatum**
image

image

### CODE REVIEW

File: packages/devtools-reps/src/launchpad/components/Console.css [hunk 1]
```diff
@@ -38,6 +38,11 @@ main {
   margin-bottom: 0.5rem;
 }
 
+.rep-element {
```
Comment (gregtatum) [inline packages/devtools-reps/src/launchpad/components/Console.css]: This aligns the styling to the webconsole's. The newlines were being ignored before.

**Comment by jasonLaster** [comment]
Nice work @gregtatum!

couple of questions
1. what does it look like if you expand?
2. should we flip the pink/blue. i believe pink is secondary... This looks good, just not sure
3. did you try getFunctionName?

**Comment by nchevobbe** [comment]
That's great Greg, thank you so much for working on this.
Regarding colors, we should ensure that they work well on a red background (which can happen if we do `console.error(err)`)

I was thinking, could we simply prevent user to expand it ?
It is an inspectable object, but I think users don't care much about it ?

For the custom formatters work, I was thinking we could have an "enhanced" icon. Basically it would be something that would tell you that a special treatment was applied to the object, and let you switch to the "raw" version if you want to inspect the object.

So here we would have the cool output 
```
✨ Error: "A complicated message"
      λ a   script.js:23
      λ b   script.js:3
```

And if you hover the ✨ you'll have a "this is an enhanced version of the logged object. Click to see it raw" (need better wording).

And then, on click, we give the user: 
```
▶︎ ✨ Error: "A complicated message"
```
which only show the error class and message. The user can then expand it to access columnNumber, stack, …

And I think we should force this view when the Error object is in another object (array, object, map, …). Not sure if relying only on MODE would be sufficient.

**Comment by tromey** [comment]
>  So here we could add an onClick on the location element, calling onViewSourceInDebugger.

If you're going this route, how much harder would it be to just fix #452?

**Comment by nchevobbe** [comment]
> If you're going this route, how much harder would it be to just fix #452?

Not so hard, we need to plug in the sourcemap service. But I think this has to be passed from the consumer, and that would require a bit of work to re-render the stack when the sourcemap is ready.

I'd say we can do it in another PR, soon after merging this one (and we would make both part of the same Reps bundle)

**Comment by jasonLaster** [comment]
Yep - I'm looking at the service for a perf bug. Happy to consult, but it's pretty straight forward.

**Comment by nchevobbe** [comment]
For the colors, here's what we are using in the console: 

function name: var(--console-output-color, currentColor)
file name: --frame-link-source
line and column numbers: --frame-link-line-color

they were carefully chosen to not clash with the different background color we can have in both light and dark mode (red, yellow, white/black)

**Comment by nchevobbe** [comment]
re embedded errors: 

image

This looks good to me. Stack is not shown in TINY mode, so it won't appear in previews, and when expanded, it takes more vertical space, but this is fine to me

**Comment by gregtatum** [comment]
I didn't get to this today, I see you approved this, what all from the feedback would you like addressed before being merged?

**Comment by nchevobbe** [comment]
Let's merge it now and address the comments in follow ups

**Comment by nchevobbe** [comment]
@gregtatum I filed a handful of follow up issues. Feel free to tackle some, or not.
I think that we should have at least clickable frames for the next reps bundle (and I'll probably work on this today).

**Comment by nchevobbe** [comment]
See https://github.com/devtools-html/devtools-core/projects/3

**Pull request merged**